Upstreamize API guide (#3212)
theforeman/foreman-documentation: guides/doc-Project_API/topics/proc_configuring-oauth.adoc

[id="configuring-oauth"]
= Configuring OAuth

> To change the OAuth settings, enter the `{foreman-installer}` with the required options.

Reviewer: ```suggestion
Use `{foreman-installer}` to change OAuth settings on {ProjectServer}.
```

Enter the following command to list all the OAuth related installer options:

[options="nowrap", subs="+quotes,attributes"]
----
# {foreman-installer} --full-help | grep oauth
----

.Enabling OAuth mapping

By default, {Project} authorizes all OAuth API requests as the built-in anonymous API administrator account.
Therefore, API responses include all {Project} data.
However, you can also specify the {Project} user that makes the request and restrict access to data to that user.

To enable OAuth user mapping, enter the following command:

[options="nowrap", subs="+quotes,attributes"]
----
# {foreman-installer} --foreman-oauth-map-users true
----

[IMPORTANT]
====
{Project} does not sign the header in an OAuth request.

> Anyone with a valid consumer key can impersonate any {Project} user.

Reviewer: This sounds like an issue at first. Can we extend this with "Ensure to keep to oAuth consumer key safe." or similar? Or maybe any other recommendation to mitigate this?

Author: I've asked for a clarification on Slack. @adamruzicka Reminder :)
